File: CompetitiveProgramming/DMOJ/CPC/CPC '21 Contest 1 P1 - AQT and Fractions/2021-08-08/Fractions.cpp
#include <bits/stdc++.h>

using namespace std;

typedef long long ll;

long long gcdEuclidean(long long a, long long b)
{
    if(a == 0)
        return b;
    return gcdEuclidean(b%a, a);
}

long long multiples(int v, long long d, long long c)
{
    if(d%v != 0)
        return c;
    return multiples(v, d/v, ++c);
}

int t; long long n, d;
int main()
{
    for(cin >> t; t--;)
    {
        cin >> n >> d;
        d /= gcdEuclidean(n, d);

        long long m2 = multiples(2, d, 0);
        long long m5 = multiples(5, d, 0);
        long long reduced = d / pow(2, m2) / pow(5, m5);
        
        if(reduced == 1)
            cout << max(m2, m5) << '\n';
        else
            cout << -1 << '\n';
    }

    return 0;
}

File: CompetitiveProgramming/DMOJ/CPC/CPC '21 Contest 1 P1 - AQT and Fractions/2021-08-08/Fractions.py
def gcdEuclidean(a, b):
    if a == 0:
        return b
    return gcdEuclidean(b%a, a)

def main():
    for _ in range(int(input())):
        n, d = map(int, input().split())
        g = gcdEuclidean(n, d)
        n /= g
        d /= g

        reduced = d
        while reduced % 2 == 0: reduced/=2
        while reduced % 5 == 0: reduced/=5

        if reduced == 1:
            count = 0
            while n % d:
                n = n * 10 % d
                count += 1
            print(count)
        else:
            print(-1)
main()